src/clients/avail/client.rs

```diff
@@ -0,0 +1,258 @@
+            error: e.into(),
+            is_transient: false,
+        })?;
+        let call = api::tx()
+            .data_availability()
+            .submit_data(BoundedVec(data.clone()));
+
+        let nonce = avail_subxt::tx::nonce(&client, &keypair)
+            .await
+            .map_err(|e| types::DAError {
+                error: e.into(),
+                is_transient: false,
+            })?;
+        let tx_progress = tx::send_with_nonce(
+            &client,
+            &call,
+            &keypair,
+            AppId(u32::try_from(self.app_id).map_err(|e| types::DAError {
+                error: e.into(),
+                is_transient: false,
+            })?),
+            nonce,
+        )
+        .await
+        .map_err(|e| types::DAError {
+            error: e.into(),
+            is_transient: false,
+        })?;
+        let block_hash = tx::then_in_block(tx_progress)
+            .await
+            .map_err(|e| types::DAError {
+                error: e.into(),
+                is_transient: false,
+            })?
+            .block_hash();
+
+        // Retrieve the data from the block hash
+        let block = client
+            .blocks()
+            .at(block_hash)
+            .await
+            .map_err(|e| types::DAError {
+                error: e.into(),
+                is_transient: false,
+            })?;
+        let extrinsics = block.extrinsics().await.map_err(|e| types::DAError {
+            error: e.into(),
+            is_transient: false,
+        })?;
+        let mut found = false;
+        let mut tx_idx = 0;
+        for ext in extrinsics.iter() {
+            let ext = ext.map_err(|e| types::DAError {
+                error: e.into(),
+                is_transient: false,
+            })?;
+            let call = ext.as_extrinsic::<SubmitData>();
+            if let Ok(Some(call)) = call {
+                if data.clone() == call.data.0 {
+                    found = true;
+                }
+            }
+            tx_idx += 1;
+        }
+
+        if !found {
+            return Err(DAError {
+                error: anyhow!("No DA submission found in block: {}", block_hash),
+                is_transient: false,
+            });
+        }
+
+        Ok(types::DispatchResponse {
+            blob_id: format!("{}:{}", block_hash, tx_idx),
+        })
+    }
+
+    async fn get_inclusion_data(
+        &self,
+        blob_id: &str,
+    ) -> Result<Option<types::InclusionData>, types::DAError> {
+        let (block_hash, tx_idx) = blob_id.split_once(':').ok_or_else(|| DAError {
+            error: anyhow!("Invalid blob ID format"),
+            is_transient: false,
+        })?;
+        let client = reqwest::Client::new();
+        let url = format!(
+            "{}/eth/proof/{}?index={}",
+            self.bridge_api_url, block_hash, tx_idx
+        );
+        let mut response: Response;
+        let mut retries = 0usize;
```

Comment:
are you sure we need retries here? we call the `get_inclusion_data` periodically, so even if the API was down for some time, the data will be retrieved eventually
it seems that retries here might just generate an unneeded load

Reply:
Left this for now, I think there may be a case where transient errors are retried more frequently than actually normal and want to avoid such a behaviour.